File: packages/dk-data-client/python/dk_data_client/modules/conditions.py
"""Condition / indication read surface.

Reads from `ind_silver.conditions` and uses the FastAPI
`/data-platform/conditions/resolve` wrapper.
"""

from __future__ import annotations

from typing import Any

from dk_data_client.modules._base import ModuleBase


class ConditionsModule(ModuleBase):
    async def resolve(self, name_or_code: str, *, hint: str | None = None) -> dict[str, Any]:
        body: dict[str, Any] = {"name_or_id": name_or_code}
        if hint:
            body["hint"] = hint
        return await self._call(
            method="conditions.resolve",
            path="/data-platform/conditions/resolve",
            args={"name_or_code": name_or_code, "hint": hint},
            http_method="POST",
            json_body=body,
        )

    async def search(self, query: str, *, limit: int = 25) -> list[dict[str, Any]]:
        params = {
            "select": "condition_id,canonical_name,icd10,mesh",
            "canonical_name": f"ilike.*{query}*",
            "limit": str(limit),
        }
        data = await self._call(
            method="conditions.search",
            path="/conditions",
            args={"query": query, "limit": limit},
            params=params,
        )
        return list(data) if isinstance(data, list) else []

File: packages/dk-data-client/typescript/src/modules/conditions.ts
/** Condition / indication read surface. */

import { ModuleBase } from "./_base.js";

export class ConditionsModule extends ModuleBase {
  async resolve(
    nameOrCode: string,
    opts: { hint?: string } = {},
  ): Promise<Record<string, unknown>> {
    const body: Record<string, unknown> = { name_or_id: nameOrCode };
    if (opts.hint) body.hint = opts.hint;
    return (await this.call({
      method: "conditions.resolve",
      path: "/data-platform/conditions/resolve",
      args: { nameOrCode, hint: opts.hint },
      httpMethod: "POST",
      jsonBody: body,
    })) as Record<string, unknown>;
  }

  async search(
    query: string,
    opts: { limit?: number } = {},
  ): Promise<Array<Record<string, unknown>>> {
    const data = (await this.call({
      method: "conditions.search",
      path: "/conditions",
      args: { query, limit: opts.limit ?? 25 },
      params: {
        select: "condition_id,canonical_name,icd10,mesh",
        canonical_name: `ilike.*${query}*`,
        limit: String(opts.limit ?? 25),
      },
    })) as Array<Record<string, unknown>>;
    return Array.isArray(data) ? data : [];
  }
}
